### PR TITLE
Fix punctuation surrounding-text cursor validation

### LINKED ISSUE #293: Punctuation surrounding-text cursor check can underflow
## Problem

The surrounding-text cursor range check uses `&&`:

```cpp
if (cursor <= 0 && cursor > length) {
    return;
}
```

The condition cannot be true. A valid cursor at position zero reaches `cursor - 1`. The cursor type is unsigned, so this operation underflows before `utf8::nextNChar()` runs.

## Expected behavior

The handler must return when the cursor is zero or greater than the text length.

## Proposed fix

Use `||` for the range check. Add a regression test that updates valid surrounding text with its cursor at position zero.

### PR DESCRIPTION
**Pull request by luojiyin1987**
## Summary

- Return when surrounding text has no character before the cursor.
- Prevent unsigned underflow from `cursor - 1`.
- Add a regression case for valid surrounding text with a zero cursor.

## Tests

- `git diff --check`
- `git clang-format --diff HEAD^`
- CMake configure and build with tests, cloud Pinyin, OpenCC, data, and tools enabled
- `ctest --test-dir build-pr294 --output-on-failure -j2`: 9/9 passed

Fixes #293

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected punctuation behavior when the cursor is at the beginning of surrounding text.
  * Chinese punctuation now commits correctly without incorrectly inferring a preceding character.

### CODE REVIEW

File: test/testpinyin.cpp [hunk 1]
```diff
@@ -587,6 +587,24 @@ void testVQuickPhraseTrigger(Instance *instance) {
...
+        ic->setCapabilityFlags(CapabilityFlag::SurroundingText);
+        instance->setCurrentInputMethod(ic, "pinyin", true);
+
+        ic->reset();
+        ic->surroundingText().setText("text", 0, 0);
+        FCITX_ASSERT(ic->surroundingText().isValid());
+        // Cursor zero has no preceding character.
+        ic->updateSurroundingText();
+    });
+}
```
[the remark this answers is not in the document]
Comment (luojiyin1987) [reply test/testpinyin.cpp:609]: Addressed in 6c7f75c. The test now sends a period and expects the Chinese full stop. All 9 tests pass.
